ArjunaJTA/jta/classes/com/arjuna/ats/jta/common/JTAEnvironmentBean.java

```diff
@@ -1225,4 +1236,127 @@ public void setNotifyCommitMarkableResourceRecoveryModuleOfCompleteBranches(
 			boolean notifyCommitMarkableResourceRecoveryModuleOfCompleteBranches) {
 		this.notifyCommitMarkableResourceRecoveryModuleOfCompleteBranches = notifyCommitMarkableResourceRecoveryModuleOfCompleteBranches;
 	}
+
+	/**
+     * Setting name of class which is taken to be used as transaction importer
+     * when a {@link javax.transaction.xa.Xid} needs to be imported as a transaction to currently running TM.<br>
+     * The classname which is set here has to implement {@link org.jboss.tm.SubordinateTransactionImporter} interface.
+     */
+    public void setSubordinateTransactionImporterClassName(String importerClassName) {
+        synchronized(this)
+        {
+            if(importerClassName == null)
+            {
+                this.subordinateTransactionImporter = null;
+            }
+            else if(!importerClassName.equals(this.subordinateTransactionImporterClassName))
+            {
+                this.subordinateTransactionImporter = null;
```

Comment:
Why do we need to lazy load the getter, if someone calls the setter there is good chance someone later on will call the getter. These setters and getters are too complicated and I would rather sacrifice a tiny perceived performance gain over code clarity.